torch/_dynamo/utils.py

```diff
@@ -386,6 +386,7 @@ def clone_tensor(x):
 
 def clone_input(x):
```

Comment:
I imagine that with fake tensors always on, we shouldn't need this at all?

This function is in the business of cloning inputs while preserving input metadata - which we should really just be using fake tensor for anyway - it's missing some of the cases that fake tensor has, and clones real tensor data unnecessarily (so we're more likely to OOM)

Reply:
ah cool I can look into https://github.com/pytorch/torchdynamo/issues/1751

To confirm: at this point, are we ok with fully killing the `use_fake_tensor` config and keeping it always-on?

Reply:
Thats correct. This is listed in the issue - https://github.com/pytorch/torchdynamo/issues/1751

We are using real_tensor_value_cache in some places, so we need to clean that up before removing `clone_input` entirely. I was hoping to tackle that in a follow up PR. But, if this seems like a big issue, we can fix it right here.

Reply:
> ah cool I can look into https://github.com/pytorch/torchdynamo/issues/1751
> 
> 
> 
> To confirm: at this point, are we ok with fully killing the `use_fake_tensor` config and keeping it always-on?

No. There are features that don't support it, critical for export and other use cases.

Reply:
But we have a PoR for how export can handle this no?